### PR TITLE
Add URL builder and error handling

### PR DESCRIPTION
**Pull request by ablwr**
Last PR! This doesn't affect the future blog tutorial, but it does make it a little easier for users (especially non-developers after their domains have been set up for them, like our own Sales team!).

This also adds an error catch. Usually Prebuilt loads and says what the problem is, but I noticed while working outside (and far away from the wifi) that it wasn't always loading but this error loads up quickly.

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,7 +1,63 @@
+/* eslint-disable react/no-unescaped-entities */
```
Comment (nienkedekker) [inline pages/index.tsx:1]: Q: is this for the `{" "}` things below?

File: components/Header.tsx [hunk 2]
```diff
@@ -27,6 +28,7 @@ const Header = ({ isTranscribing, owner, token }: Props) => {
             <span className={styles.alertStatus}>Not owner</span>
           )}
           {isTranscribing && <span>Transcribing</span>}
+          {hasError && <span className={styles.errorStatus}>{hasError}</span>}
```
Comment (lizadaily) [inline components/Header.tsx:31]: Sorry for nit, but I would actually suggest just calling this `error`. `hasError` makes it sound like a bool, when it contains the error string itself.

File: pages/[domain]/[room].tsx [hunk 3]
```diff
@@ -50,8 +52,16 @@ const Room: NextPage = ({}) => {
     });
     setCallFrame(newCallFrame);
 
-    newCallFrame.join({
-      url: url,
+    try {
+      newCallFrame.join({
+        url: url,
+      });
+    } catch {
```
Comment (lizadaily) [inline pages/[domain]/[room].tsx:59]: Is it worth including the root error here in what we re-throw as well for context? Or is the room not existing the only reason we expect this to ever fail?

File: pages/index.tsx [hunk 4]
```diff
@@ -1,7 +1,63 @@
...
+
+function Form() {
+  let [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  let [link, setLink] = useState<string>("");
+
+  const buildUrl = (ev: any) => {
+    ev.preventDefault(); // don't redirect the page
+    const source = ev.target;
+    const link = `${window.location.href}${source.domain.value}/${source.room.value}?t=${source.token.value}`;
+    console.log(link);
```
Comment (lizadaily) [inline pages/index.tsx:18]: Is this intended to stay or is it a temp debug log? (suggest adding a tiny bit more context to the log string if the former)

**Review by lizadaily — APPROVED** [review]

**Pull request merged**